### PR TITLE
Issue 216

### LINKED ISSUE #216: Update CI/CD, add stronger styling/autoformat
- [x] Standardize testing https://tox.wiki/en/latest/
- [x] Flake8 enforce double quotes https://github.com/zheller/flake8-quotes/
- [x] Flake8 enforce import order https://github.com/PyCQA/flake8-import-order
- [x] ~Autoformat https://github.com/psf/black/actions/runs/17913292/workflow~

### PR DESCRIPTION
**Pull request by jsstevenson**
close #216 

Style/CI
- Flake8 max line length -> 88 (eg https://news.ycombinator.com/item?id=17155228, https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#line-length)
- Flake8 enforce single -> double quotes
- Flake8 enforce PEP8 import groups https://www.python.org/dev/peps/pep-0008/#id21
- Flake8 enforce type hints
- Mypy evaluate types
- Tox (run tests in an "installed" environment rather than main environment, run all tests at once). I rewound some of the code to run the fake data fixtures back, and I'd like to revisit it once we have actual fake input data.
- Remove dynamodb_revisions (we don't need to provide DB migration-like information for the normalizers anymore, just for their upstream effects on the metakb, also a bunch of the migration scripts don't work anymore. Also it's still a 0.x release)
- Clean up root directory (remove unused files, consolidate where possible)
- Remove coveralls (unless we want to use it)
- Add beginnings of a Makefile (targets for run, clean, and rebuilding requirements.txt)

Also some internal stuff
- Update some docstrings
- Fix setup.cfg package pins

Looked at doing but didn't
- [Autoformat](https://black.readthedocs.io/en/stable/) (though I'd be down)
- More aggressive docstring checks -- there are [options](https://github.com/terrencepreilly/darglint) but we might have to switch to a different docstring syntax
- Tried to get an "is requirements/dev.txt up to date w/ the Pipfile" test going in Tox but it seems like requirements.txt generation can look slightly different between systems so that's off the table unless there's a way to check for being up to date other than pure file equality

### CODE REVIEW

File: therapy/etl/hemonc.py [hunk 1]
```diff
@@ -76,26 +80,26 @@ def _get_concepts(self) -> Tuple[Dict, Dict, Dict]:
         brand_names = {}  # hemonc id -> brand name
         conditions = {}  # hemonc id -> condition name
 
-        concepts_file = open(self._src_files[0], 'r')
+        concepts_file = open(self._src_files[0], "r")
```
[the remark this answers is not in the document]
Comment (jsstevenson) [reply therapy/etl/hemonc.py:83]: Always pro-more type annotations!

Static analysis can infer the types of a few of these -- if you initialize a dict with a specific key and value, it assumes all additional members will match that pattern and will throw an error if not. However, it breaks down when interfacing with external libraries that don't have type declarations yet (like owlready in NCIt) and won't even try to infer, so these latter additions are helpful

File: .github/workflows/github-actions.yml [hunk 2]
```diff
@@ -8,7 +8,7 @@ jobs:
             AWS_SECRET_ACCESS_KEY: ${{ secrets.DUMMY_AWS_SECRET_ACCESS_KEY }}
             AWS_DEFAULT_REGION: us-east-2
             AWS_DEFAULT_OUTPUT: text
-            THERAPY_NORM_DB_URL: http://localhost:8000
+            THERAPY_NORM_DB_URL: http://localhost:8001
```
[the remark this answers is not in the document]
Comment (jsstevenson) [reply .github/workflows/github-actions.yml:11]: My thinking was that you could keep live databases on 8000 and run databases with fake data on 8001, but the CI pipeline can only run tests against the latter

File: .flake8 [hunk 3]
```diff
@@ -1,10 +1,21 @@
...
+  analysis/*
+  codebuild/*
   therapy/version.py
+inline-quotes = "
+import-order-style = pep8
+application-import-names =
+  therapy
+  tests
+per-file-ignores =
+  tests/*:ANN001, ANN2
```
[the remark this answers is not in the document]
Comment (jsstevenson) [reply .flake8:20]: Yes that's much better

**Review by korikuzma — APPROVED** [review]
I think maybe using a better variable name than `Params` (in schemas) would be nice, but not a huge issue if you just want to stick with it

**Review by korikuzma — APPROVED** [review]
💯

**Pull request merged**